libpod/pod.go

```diff
@@ -28,14 +32,43 @@ func (p *Pod) Name() string {
 	return p.name
 }
 
-// Creates a new pod
-func newPod() (*Pod, error) {
+// Labels returns the pod's labels
+func (p *Pod) Labels() map[string]string {
+	labels := make(map[string]string)
+	for key, value := range p.labels {
+		labels[key] = value
+	}
+
+	return labels
+}
+
+// Creates a new, empty pod
+func newPod(lockDir string) (*Pod, error) {
 	pod := new(Pod)
 	pod.id = stringid.GenerateNonCryptoID()
-	pod.name = pod.id // TODO generate human-readable name here
+	pod.name = namesgenerator.GetRandomName(0)
 
 	pod.containers = make(map[string]*Container)
 
+	// TODO: containers and pods share a locks folder, but not tables in the
+	// database
+	// As the locks are 256-bit pseudorandom integers, collision is unlikely
+	// But it's something work looking into
+
+	// Path our lock file will reside at
+	lockPath := filepath.Join(lockDir, pod.id)
+	// Ensure there is no conflict - file does not exist
+	_, err := os.Stat(lockPath)
+	if err == nil || !os.IsNotExist(err) {
+		return nil, errors.Wrapf(ErrCtrExists, "lockfile for pod ID %s already exists", pod.id)
```

Comment:
Is this an error, are we sure that the lock files are always cleaned up?  If the file does exist would it be a problem if we used it?  I would think it better to wrap the "err" in the case where it does not exists?

Reply:
I don't think it's possible to clean up lockfiles directly, given we can't close their FDs. As such it might be better to remove these checks and move them into tmpfs so they clean up on reboot if nothing else

Reply:
SGTM, also might be faster since writing to a tmpfs should be quicker then physical storage.  I was thinking more about a container pid being reused. Remotely possible.

Reply:
Removing the checks should also handle that